File: 프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.py
def solution(stones, k):
    answer = 0
    left = 0
    right = max(stones)
    while left <= right:
        mid = (left + right) // 2
        possible = True
        count = 0
        for stone in stones:
            if stone < mid:
                count += 1
                if count >= k:
                    possible = False
                    break
            else:
                count = 0
        if possible:
            answer = mid
            left = mid + 1
        else:
            right = mid - 1
    
    return answer

File: 프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.java
class Solution {
    public int solution(int[] stones, int k) {
        int maxStone = 0;
        for (int stone : stones) if (stone > maxStone) maxStone = stone;
        int left = 1;
        int right = maxStone;
        
        while (left <= right) {
            int mid = (left + right) / 2;
            int count = 0;
            boolean possible = true;
            for (int stone : stones) {
                if (stone < mid) count ++;
                else count = 0;
                if (count >= k) {
                    right = mid-1;
                    possible = false;
                    break;
                }
            }
            if (possible) left = mid+1;
        }
        return right;
    }
}